src/realtime/pubsub/consumer_map.rs

```diff
@@ -1,11 +1,18 @@
-use std::collections::{HashMap, HashSet};
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
 
-use super::{consumer::EventConsumer, topic::Topic};
+use super::topic::Topic;
+use crate::realtime::socket::Socket;
+
+pub type EventConsumer = Arc<Socket>;
```

Comment:
Replacing all occurrences of `EventConsumer` with `Arc<Socket>` is better because rust treats `EventConsumer` as distinct to `Arc<Socket>`, which is annoying.